packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -479,7 +479,7 @@ void main() {
     return test.main;
   });
   WebSocket.connect(server).then((WebSocket socket) {
-    socket.map(JSON.decode).pipe(channel.sink);
+    socket.map((dynamic x) => JSON.decode(x)).pipe(channel.sink);
```

Comment:
This code seems dangerous. What if the socket sends a binary packet?

Reply:
I think we should assert that x is String.

Reply:
If socket sends a binary packet you will get runtime type exception from `JSON.decode` - it only deals with strings. I think this code is written to talk to the known end point (communication between test runner and tests themselves) so we know that we only need to expect strings.

Reply:
@Hixie added assert

Reply:
In Dart 2, you will get a runtime type error from the dynamic invocation since you can't invoke `json.decode` with a non-string.

Also, use lower-case constants: `json`, not `JSON`.